libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContent.java

```diff
@@ -84,4 +84,37 @@ XContentParser createParser(NamedXContentRegistry xContentRegistry,
      */
     XContentParser createParser(NamedXContentRegistry xContentRegistry,
             DeprecationHandler deprecationHandler, Reader reader) throws IOException;
+
+    /**
+     * Creates a parser over the provided string content.
+     */
+    XContentParser createParser(NamedXContentRegistry xContentRegistry,
```

Comment:
it is tempting to  create a builder for this instead of the factor methods, but it would be a big refactoring..